src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,12 +1,69 @@
-// (C) 2007-2019 GoodData Corporation
+// (C) 2020 GoodData Corporation
 import * as React from "react";
+import without = require("lodash/without");
+import * as classNames from "classnames";
+import { Execution } from "@gooddata/typings";
+import { stringToFloat } from "../../../helpers/utils";
+import { IntlWrapper } from "../../core/base/IntlWrapper";
+import { IntlTranslationsProvider, ITranslationsComponentProps } from "../../core/base/TranslationsProvider";
+import PushpinSizeLegend from "./legends/PushpinSizeLegend";
+import { getGeoData, calculateAverage, getFormatFromExecutionResponse } from "../../../helpers/geoChart";
+import { IGeoConfig, IGeoData } from "../../../interfaces/GeoChart";
+import { TOP } from "../../visualizations/chart/legend/PositionTypes";
+import { isTwoDimensionsData } from "../../../helpers/executionResultHelper";
 
-export interface IChartLegendProps {
-    domProps: any;
+export interface IGeoChartLegendRendererProps {
+    config: IGeoConfig;
+    execution: Execution.IExecutionResponses;
+    locale: string;
+    position?: string;
 }
+export default function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): JSX.Element {
+    const {
+        execution: { executionResult, executionResponse },
+        config: { mdObject: { buckets = [] } = {} },
+        locale,
+        position = TOP,
+    } = props;
+    const geoData: IGeoData = getGeoData(buckets, executionResponse.dimensions);
+    const { data } = executionResult;
+    const classes = ["geo-legend", "s-geo-legend", `position-${position}`];
+    const hasSize: boolean = geoData.size !== undefined;
 
-export default class GeoChartLegendRenderer extends React.Component<IChartLegendProps> {
-    public render(): React.ReactElement {
+    if (!hasSize || !isTwoDimensionsData(data)) {
         return null;
     }
+    return (
+        <div className={classNames(...classes)}>
+            {hasSize &&
+                renderPushpinSizeLegend(
+                    data[geoData.size.index].map(stringToFloat),
+                    getFormatFromExecutionResponse(geoData.size.index, executionResponse),
+                    locale,
+                )}
+        </div>
+    );
```

Comment:
how about this?
```
    ....
    const { size: { index: sizeIndex } } = geoData;
    ....
    const sizesInNumber = data[sizeIndex].map(stringToFloat);
    const format = getFormatFromExecutionResponse(geoData.size.index, executionResponse);
    return (
        <div className={classNames(...classes)}>
            {
                renderPushpinSizeLegend(
                    sizesInNumber,
                    format,
                    locale,
                )}
        </div>
    );
```

Reply:
- it will throw error if geoData don't have size property. so it could:
       const { size } = geoData;
- About format, we also have format for Color, and Size.

Reply:
I see we already check `!size` in if condition, we should remove `{ size &&`, shouldn't we?

Reply:
Because Geo have three Legend type, so we still need Size condition to show/hide legend